src/main/java/net/minecraftforge/common/lighting/LightingEngine.java

```diff
@@ -0,0 +1,630 @@
+    private EnumSkyBlock lightType;
+
+    //Iteration state data
+    //Cache position to avoid allocation of new object each time
+    private final MutableBlockPos curPos = new MutableBlockPos();
+    private PooledLongQueue curQueue;
+    private Chunk curChunk;
+    private long curData;
+
+    //Cached data about neighboring blocks (of tempPos)
+    private boolean isNeighborDataValid = false;
+    private final Chunk[] neighborsChunk = new Chunk[EnumFacing.VALUES.length];
+    private final MutableBlockPos[] neighborsPos = new MutableBlockPos[EnumFacing.VALUES.length];
+    private final int[] neighborsLight = new int[EnumFacing.VALUES.length];
+
+    public LightingEngine(final World world)
+    {
+        this.world = world;
+        this.profiler = world.theProfiler;
+
+        for (int i = 0; i < EnumSkyBlock.values().length; ++i)
+        {
+            this.queuedLightUpdates[i] = new PooledLongQueue();
+        }
+
+        for (int i = 0; i < this.queuedDarkenings.length; ++i)
+        {
+            this.queuedDarkenings[i] = new PooledLongQueue();
+        }
+
+        for (int i = 0; i < this.queuedBrightenings.length; ++i)
+        {
+            this.queuedBrightenings[i] = new PooledLongQueue();
+        }
+
+        for (int i = 0; i < this.neighborsPos.length; ++i)
+        {
+            this.neighborsPos[i] = new MutableBlockPos();
+        }
+    }
+
+    /**
+     * Schedules a light update for the specified light type and position to be processed later by {@link #procLightUpdates(EnumSkyBlock)}
+     */
+    public void scheduleLightUpdate(final EnumSkyBlock lightType, final BlockPos pos)
+    {
+        this.scheduleLightUpdate(lightType, posToLong(pos));
+    }
+
+    /**
+     * Schedules a light update for the specified light type and position to be processed later by {@link #procLightUpdates()}
+     */
+    private void scheduleLightUpdate(final EnumSkyBlock lightType, final long pos)
+    {
+        final PooledLongQueue queue = this.queuedLightUpdates[lightType.ordinal()];
+
+        queue.add(pos);
+
+        //make sure there are not too many queued light updates
+        if (queue.size() >= MAX_SCHEDULED_COUNT)
+        {
+            this.procLightUpdates(lightType);
+        }
+    }
+
+    /**
+     * Calls {@link #procLightUpdates(EnumSkyBlock)} for both light types
+     */
+    public void procLightUpdates()
+    {
+        this.procLightUpdates(EnumSkyBlock.SKY);
+        this.procLightUpdates(EnumSkyBlock.BLOCK);
+    }
+
+    /**
+     * Processes light updates of the given light type
+     */
+    public void procLightUpdates(final EnumSkyBlock lightType)
+    {
+        final PooledLongQueue queue = this.queuedLightUpdates[lightType.ordinal()];
+
+        if (queue.isEmpty())
+        {
+            return;
+        }
+
+        //renderer accesses world unsynchronized, don't modify anything in that case
+        if (this.world.isRemote && !Minecraft.getMinecraft().isCallingFromMinecraftThread())
```

Comment:
Referencing client side code is fine, you just have to be sure not to load the class/method thats references it.
Or reflectively loop through the class that references it. ForgeHooks is fine because there should be no reason for people to reflect through. In this case however ass lighting is done both sides you'd probably want to wrap this around a FMLHandler